File: xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 9/C#/Html5CanvasDrawing.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using NUnit.Framework;
using System.Collections;
using System.Xml.Linq;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Support;
using OpenQA.Selenium.Interactions;
using OpenQA.Selenium.Support.UI;
using System.Drawing.Imaging;

namespace HTML5Tests
{
    [TestFixture]
    public class Html5CanvasDrawing
    {
        IWebDriver driver;

        [SetUp]
        public void TestSetup()
        {
            // Create a instance of the Chrome driver using IWebDriver Interface
            driver = new ChromeDriver();
        }

        [TestCase]
        public void TestHtml5CanvasDrawing()
        {
            try
            {
                driver.Navigate().GoToUrl("http://dl.dropbox.com/u/55228056/html5canvasdraw.html");

                //Get the HTML5 Canvas Element
                IWebElement canvas = driver.FindElement(By.Id("imageTemp"));
                //Select the Pencil Tool
                SelectElement drawtool = new SelectElement(driver.FindElement(By.Id("dtool")));
                drawtool.SelectByText("Pencil");

                //Create a Action Chain for Draw a shape on Canvas
                Actions builder = new Actions(driver);
                builder.ClickAndHold(canvas).MoveByOffset(10, 50).
                                             MoveByOffset(50, 10).
                                             MoveByOffset(-10, -50).
                                             MoveByOffset(-50, -10).Release().Perform();

                //Get a screenshot of Canvas element after Drawing and compare it to the base version
                Screenshot ss = ((ITakesScreenshot)driver).GetScreenshot();
                string screenshot = ss.AsBase64EncodedString;
                byte[] screenshotAsByteArray = ss.AsByteArray;
                ss.SaveAsFile(@"c:\tmp\post.png", ImageFormat.Png);
            }
            catch (Exception e)
            {
                Assert.Fail("Test Failed due to exception '" + e.Message + "'");
            }
        }

        [TearDown]
        public void TestCleanUp()
        {
            // Close the browser   
            driver.Quit();
        }
    }
}

File: xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 9/Ruby/html5_canvas_drawing.rb
require 'rubygems'
require 'selenium-webdriver'
require 'test/unit'

class Html5CanvasDrawing < Test::Unit::TestCase
	def setup
		@driver = Selenium::WebDriver.for :chrome
		@driver.get 'http://dl.dropbox.com/u/55228056/html5canvasdraw.html'
		@verification_errors = []
	end
	
	def test_html5_canvas_drawing
		#Get the HTML5 Canvas Element
		canvas = @driver.find_element :id=> 'imageTemp'
		
		#Select the Pencil Tool
		drawtool = Selenium::WebDriver::Support::Select.new(@driver.find_element(:id => 'dtool'))
		drawtool.select_by :text, 'Pencil'
		
		#Create a Action Chain for Draw a shape on Canvas
		@driver.action.click_and_hold(canvas).move_by(10, 50).
									 move_by(50,10).
									 move_by(-10,-50).
									 move_by(-50,-10).release().perform()
		
		#Get a screenshot of Canvas element after Drawing 
		@driver.save_screenshot('c:\\tmp\\post.png')
	end
	
	def teardown
		@driver.quit
		assert_equal [], @verification_errors
	end
end